tests/test_async.py

```diff
@@ -156,3 +156,40 @@ def echo_client(message, loop):
 
         self.assertEqual(server_messages, ['Hello, World!', 'Goodbye, World!'])
         self.assertEqual(client_messages, ['Hello, World!', 'Goodbye, World!'])
+
+
+class AsyncSubprocessTests(unittest.TestCase):
+    def setUp(self):
+        asyncio.set_event_loop_policy(EventLoopPolicy())
+        self.loop = asyncio.get_event_loop()
+
+    def tearDown(self):
+        asyncio.set_event_loop_policy(None)
+        self.loop.close()
+
+    def test_subprocess(self):
+        @asyncio.coroutine
+        def list_dir():
+            self.proc = yield from asyncio.create_subprocess_shell(
+                'ls',
+                stdout=asyncio.subprocess.PIPE,
+            )
+
+            entries = set()
+            line = yield from self.proc.stdout.readline()
+            while line:
+                entries.add(line.decode('utf-8').strip())
```

Comment:
It *might* be more correct to use `locale.getpreferredencoding()` as the encoding here, because the output of `ls` will be encoded as whatever the current locale specifies. On the other hand, practically everything uses UTF-8 as the default encoding nowadays, especially on Apple platforms, so it's unlikely to matter in practice.

Reply:
I think you're right - this is one of those "technically correct, but not in a way that is actually ever helpful" thing. Is there any platform other than Apple where this code could run, even in theory?

Reply:
Yes, Linux. clang, libobjc, libclosure (blocks) and Core Foundation are open-source and work on Linux, and there is [GNUstep](http://gnustep.org/). Not exactly a common scenario, but at least one person has attempted to run Rubicon on Linux in the past (#13).

It's probably not worth worrying about though - the likelihood of someone using Rubicon on Linux with a non-UTF-8 locale is *very* small, in the Linux world UTF-8 is basically standard as far as I know. And even if someone wants to use such a setup, the only thing that would break is a unit test.

Alternatively, just don't decode the output and compare against bytes. That will work for all ASCII-based encodings.